### PR TITLE
Automatically create certificates if required

### PR DESCRIPTION
**Pull request by tgraf**
Signed-off-by: Thomas Graf <thomas@cilium.io>

### CODE REVIEW

File: cilium-etcd-operator.yaml [hunk 1]
```diff
@@ -65,6 +67,11 @@ rules:
 - apiGroups: ["extensions"]
   resources: ["deployments"]
   verbs: ["create", "get","list","patch","update"]
+- apiGroups:
+  - ""
+  resources:
+  - secrets
+  verbs: ["get", "create", "update"]
```
Comment (aanm) [inline cilium-etcd-operator.yaml]: Do we need the "update"?

Comment (tgraf) [reply cilium-etcd-operator.yaml]: Technically not required, removed

File: cilium-etcd-operator.yaml [hunk 2]
```diff
@@ -14,14 +14,16 @@ spec:
       hostNetwork: true
       containers:
       - name: cilium-etcd-operator
-        image: docker.io/cilium/cilium-etcd-operator:1.0
+        image: docker.io/cilium/cilium-etcd-operator:1.1
         imagePullPolicy: Always
```
Comment (aanm) [inline cilium-etcd-operator.yaml]: Change the `imagePullPolicy: IfNotPresent`

File: run.sh [hunk 3]
```diff
@@ -24,6 +32,44 @@ API_TOKEN_PATH=${API_TOKEN_PATH:-/var/run/secrets/kubernetes.io/serviceaccount/t
...
+
+	mv peer.pem peer.crt
+	mv peer-key.pem peer.key
+	mv ca.pem peer-ca.crt
+
+	rm *.csr ca-key.pem
+
+	kubectl create secret generic -n kube-system cilium-etcd-peer-tls --from-file=peer-ca.crt --from-file=peer.crt --from-file=peer.key
+	kubectl create secret generic -n kube-system cilium-etcd-server-tls --from-file=server-ca.crt --from-file=server.crt --from-file=server.key
+	kubectl create secret generic -n kube-system cilium-etcd-client-tls --from-file=etcd-client-ca.crt --from-file=etcd-client.crt --from-file=etcd-client.key
```
Comment (aanm) [inline run.sh:73]: The certificates were uploaded to kubernetes, I think it's reasonable paranoid to delete them from this container.

Comment (tgraf) [reply run.sh:73]: Good point

**Pull request merged**